src/plugins/unified_histogram/public/layout/hooks/use_lens_suggestions.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { DataView } from '@kbn/data-views-plugin/common';
+import { AggregateQuery, isOfAggregateQueryType, Query } from '@kbn/es-query';
+import { LensSuggestionsApi, Suggestion } from '@kbn/lens-plugin/public';
+import { isEqual } from 'lodash';
+import { useEffect, useMemo, useRef, useState } from 'react';
+
+export const useLensSuggestions = ({
+  dataView,
+  query,
+  originalSuggestion,
+  isPlainRecord,
+  columns,
+  lensSuggestionsApi,
+  onSuggestionChange,
+}: {
+  dataView: DataView;
+  query?: Query | AggregateQuery;
+  originalSuggestion?: Suggestion;
+  isPlainRecord?: boolean;
+  columns?: string[];
+  lensSuggestionsApi: LensSuggestionsApi;
+  onSuggestionChange?: (suggestion: Suggestion | undefined) => void;
+}) => {
+  const suggestions = useMemo(() => {
+    const context = {
+      dataViewSpec: dataView?.toSpec(),
+      fieldName: '',
+      contextualFields: columns,
+      query: query && isOfAggregateQueryType(query) ? query : undefined,
+    };
+    const lensSuggestions = isPlainRecord ? lensSuggestionsApi(context, dataView) : undefined;
+    const firstSuggestion = lensSuggestions?.length ? lensSuggestions[0] : undefined;
+    const restSuggestions = lensSuggestions?.filter((sug) => {
```

Comment:
Marco I thought about it but I decided against it as I would like Lens to return all suggestions and consumers will decide what they want to depict and what not.

Reply:
I do not see how that can conflict with my proposal, the consumer is still in power to decide either way.
It is just cleaner, from my point of view, to make Lens decide on the filtering as it could produce more suggestions of other visualization types that way.

Reply:
If you are ok with it let's do it on a follwup. We really need to merge this PR :)